pkg/api/login.go

```diff
@@ -210,7 +210,21 @@ func (hs *HTTPServer) LoginPost(c *models.ReqContext) response.Response {
 	err := bus.Dispatch(c.Req.Context(), authQuery)
 	authModule = authQuery.AuthModule
 	if err != nil {
-		resp = response.Error(401, "Invalid username or password", err)
+		var msg string
+
+		if errors.Is(err, login.ErrTooManyLoginAttempts) {
+			msg = "Too many consecutive incorrect login attempts for user - login for user temporarily blocked"
+		}
+
+		if errors.Is(err, login.ErrInvalidCredentials) {
+			msg = "Invalid username or password"
+		}
+
+		if errors.Is(err, models.ErrUserNotFound) {
+			msg = "User not found"
+		}
```

Comment:
Don't tell a potentially hostile user that the username isn't valid.
 https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Authentication_Cheat_Sheet.md#authentication-and-error-messages